src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/non_max_suppression_gpu_ref.cl

```diff
@@ -530,34 +532,31 @@ KERNEL (non_max_suppression_ref_stage_3)(
 #endif
 
     unroll_for (int i = 0; i < outputIdx; i++) {
-        const int offset = 3 * i;
-        output[offset + 0] = sortedBoxList[i].batchId;
-        output[offset + 1] = sortedBoxList[i].classId;
-        output[offset + 2] = sortedBoxList[i].boxId;
+        output[OUTPUT_GET_INDEX(i, 0, 0, 0)] = sortedBoxList[i].batchId;
+        output[OUTPUT_GET_INDEX(i, 1, 0, 0)] = sortedBoxList[i].classId;
+        output[OUTPUT_GET_INDEX(i, 2, 0, 0)] = sortedBoxList[i].boxId;
     }
 
     // Padding
     unroll_for (int i = outputIdx; i < OUTPUT_NUM; i++) {
         const int offset = 3 * i;
-        output[offset + 0] = -1;
-        output[offset + 1] = -1;
-        output[offset + 2] = -1;
+        output[OUTPUT_GET_INDEX(i, 0, 0, 0)] = -1;
```

Comment:
It is not always b,f,y,x. Is this validated?

Reply:
don't understand, what do you mean? 
we discussed that working with format is encapsulated in macroses, is my understanding is incorrect?
If you mean if NMS can be only 4-dimensional - it is

Reply:
>> If you mean if NMS can be only 4-dimensional - it is
Yes I was asking this. If it is always true, its okay.

Reply:
Yes
1 : boxes - tensor of type T and shape [num_batches, num_boxes, 4] with box coordinates.

2 : scores - tensor of type T and shape [num_batches, num_classes, num_boxes] with box scores.

Ouput:
selected_indices - tensor of type output_type and shape [number of selected boxes, 3] containing information about selected boxes as triplets [batch_index, class_index, box_index].